sklearn/model_selection/tests/test_search.py

```diff
@@ -319,14 +304,38 @@ def test_grid_search_one_grid_point():
     assert_array_equal(clf.dual_coef_, cv.best_estimator_.dual_coef_)
 
 
+def test_grid_search_when_param_grid_includes_range():
+    # Test that the best estimator contains the right value for foo_param
+    clf = MockClassifier()
+    grid_search = None
+    if PY3_OR_LATER:
+        grid_search = GridSearchCV(clf, {'foo_param': range(1, 4)}, verbose=3)
+    else:
+        grid_search = GridSearchCV(clf, {'foo_param': xrange(1, 4)}, verbose=3)
+    # make sure it selects the smallest parameter in case of ties
+    old_stdout = sys.stdout
+    sys.stdout = StringIO()
+    grid_search.fit(X, y)
+    sys.stdout = old_stdout
```

Comment:
You are right for the sake of the test it's unnecessary, test has been refactored. Thank you.